src/utils/sources-tree/utils.js

```diff
@@ -37,8 +37,7 @@ export function isDirectory(url: Object) {
   );
 }
 
-export function getExtension(source: Object): string {
-  const url = source.get ? source.get("url") : source.url;
+export function getExtension(url: string = ""): string {
```

Comment:
Our snapshot tests are written in such a way that having the url as undefined triggers a ton of snapshot errors, this is why the empty string default.